TaskfMRIAnalysis/TaskfMRIAnalysis.sh

```diff
@@ -39,56 +39,92 @@
-#  Usage Description Function
-# --------------------------------------------------------------------------------
-
-script_name=$(basename "${0}")
+# ------------------------------------------------------------------------------
+#  Check that HCPPIPEDIR is defined and Load Function Libraries
+# ------------------------------------------------------------------------------
 
-show_usage() {
-	cat <<EOF
+pipedirguessed=0
+if [[ "${HCPPIPEDIR:-}" == "" ]]
+then
+    pipedirguessed=1
+    export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
 
-${script_name}: Run TaskfMRIAnalysis pipeline
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/fsl_version.shlib"          # Function for getting FSL version
 
-Usage: ${script_name} [options]
 
-Usage information To Be Written
+# --------------------------------------------------------------------------------
+#  Usage Description Function
+# --------------------------------------------------------------------------------
 
-EOF
+#this function gets called by opts_ParseArguments when --help is specified
+function usage()
+{
+    #header text
+    echo "
+$log_ToolName: Run TaskfMRIAnalysis pipeline for a subject. Pipeline will run Level1 (scan-level) analyses, and Level2 (single subject-level) analysis as specified.
+
+Usage: $log_ToolName arguments...
+[ ] = optional; < > = user supplied value
+"
+    #automatic argument descriptions
+    opts_ShowArguments
+    
+    #do not use exit, the parsing code takes care of it
 }
 
-# Allow script to return a Usage statement, before any other output or checking
-if [ "$#" = "0" ]; then
-    show_usage
-    exit 1
-fi
-
 # ------------------------------------------------------------------------------
-#  Check that HCPPIPEDIR is defined and Load Function Libraries
+#  Parse Command Line Options
 # ------------------------------------------------------------------------------
 
-if [ -z "${HCPPIPEDIR}" ]; then
-  echo "${script_name}: ABORTING: HCPPIPEDIR environment variable must be set"
-  exit 1
-fi
+#arguments to opts_Add*: switch, variable to set, name for inside of <> in help text, description, [default value if AddOptional], [compatibility flag, ...]
+#help info for option gets printed like "--foo=<$3> - $4"
+opts_AddMandatory '--study-folder' 'Path' '/path/to/study/folder' "directory containing imaging data for all subjects"
+opts_AddMandatory '--subject' 'Subject' 'SubjectID' ""
+opts_AddMandatory '--lvl1tasks' 'LevelOnefMRINames' 'tfMRI_TaskName_RL@tfMRI_TaskName_LR' "List of subdirectories containing each time series image for the TaskName task. Multiple scan subdirectories should be provided as a single string separated by '@' character." #Assumes these subdirectories are located in the SubjectID/MNINonLinear/Results directory. Also assumes that timeseries image filename begins with this string.
```

Comment:
For the description, I would use a terminology more in keeping with other scripts.  i.e., it's a list of "fMRI scan names" , not "subdirectories" (which could imply that providing a path as part of the name is appropriate).

Reply:
Also, I suggest we switch to using AP and PA as the examples (consistent with HCP-LS), not the RL/LR from HCP-YA.

Reply:
Why does that matter?

Reply:
It is simply a more pertinent example for our current data/processing, and also how every one is acquiring their data.  (No one is acquiring RL/LR).

Reply:
> For the description, I would use a terminology more in keeping with other scripts. i.e., it's a list of "fMRI scan names" , not "subdirectories" (which could imply that providing a path as part of the name is appropriate).

The only concern that I have is that this does refer specifically to the value of the subdirectory name. If we say "list of fMRI scan names", are users going to set the value to *tfMRI_TASK_AP_Atlas_YadaYada.dtseries.nii*?

Reply:
Is ${fMRIName} not informative enough to folks?  fMRI base name?

Reply:
The convention I have been using is that something that takes a filename says "file" or something more descriptive like "timeseries", while a string used in naming is "name" or "string".  In the fairly new ExtractFromMRFIXConcat.sh, I used `--multirun-fix-names=<run1@run2...> - list of run names used in MR FIX, in the SAME ORDER, separated by @s`, sidestepping the use of realistic names in the example, but perhaps not as obvious, and made more tolerable by the fact that it is a copy of one of the MR FIX arguments.

Reply:
Changed example to 'ScanName1@ScanName2'